Interop/Indexer/src/main/kotlin/org/jetbrains/kotlin/native/interop/indexer/NativeIndex.kt

```diff
@@ -265,10 +265,16 @@ object CBoolType : BoolType()
 
 object ObjCBoolType : BoolType()
 
-data class IntegerType(val size: Int, val isSigned: Boolean, val spelling: String) : PrimitiveType
+class IntegerType(val size: Int, val isSigned: Boolean, spelling: String) : PrimitiveType {
```

Comment:
Not sure if removing `data` (including custom `hashCode`/`equals`/`toString`) is desirable here.
Can we fix the spelling on call site of these constructors, i.e. in `convertUnqualifiedPrimitiveType`?
Does `clang_getCanonicalType` remove `const` modifier from type?

Reply:
> Not sure if removing data (including custom hashCode/equals/toString) is desirable here.
Yup, it broke `interop_objc_smoke` test :)

Reply:
> Can we fix the spelling on call site of these constructors, i.e. in convertUnqualifiedPrimitiveType?

Yes.
> Does clang_getCanonicalType remove const modifier from type?

No. There is no C API for dropping `const` qualifier. It is possible to write a wrapper around  `Qualifiers::removeConst` but i don't think it's any better than string manipulation in this case.